llvm/lib/Target/X86/MCTargetDesc/X86AsmBackend.cpp

```diff
@@ -1546,3 +1547,32 @@ MCAsmBackend *llvm::createX86_64AsmBackend(const Target &T,
     return new ELFX86_X32AsmBackend(T, OSABI, STI);
   return new ELFX86_64AsmBackend(T, OSABI, STI);
 }
+
+namespace {
+class X86ELFStreamer : public MCELFStreamer {
+public:
+  X86ELFStreamer(MCContext &Context, std::unique_ptr<MCAsmBackend> TAB,
```

Comment:
How about COFF streamer?

Reply:
It's in `llvm/lib/Target/X86/MCTargetDesc/X86WinCOFFStreamer.cpp`, which doesn't support Intel JCC Erratum.

There is some way to split X86AsmBackend.cpp into X86ELFStreamer.cpp, but that would be a quite large refactoring and otherwise do not yield a noticeable gain.

Reply:
IIUC, Intel JC Erratum supports COFF now

```
llvm-mc -filetype=obj -triple  x86_64-pc-win32 --x86-align-branch-boundary=32 --x86-align-branch=call+jmp+indirect+ret+jcc ./llvm/test/MC/X86/align-branch-single.s | llvm-objdump -d --no-show-raw-insn -
```

though I didn't add test for it.  Would this patch drop the support for COFF?

Reply:
I added `X86_MC::emitInstruction` to retain the support for COFF.